### PR TITLE
Serialize only __root__ value

### PR DESCRIPTION
**Pull request by dmontagu**
## Change Summary

Converts `.json()` to return the contents of the `__root__` key for models with a custom root.

## Related issue number

#730

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.rst` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

File: docs/examples/custom_root_field.py [hunk 1]
```diff
@@ -9,19 +9,27 @@ class Pets(BaseModel):
 print(Pets(__root__=['dog', 'cat']))
 # > Pets __root__=['dog', 'cat']
 
-print(Pets.parse_obj(__root__=['dog', 'cat']))
```
Comment (dmontagu) [inline docs/examples/custom_root_field.py]: this was actually a bug

File: docs/examples/custom_root_field.json [hunk 2]
```diff
@@ -1,11 +0,0 @@
-{
-  "definitions": {
-    "Pets": {
-      "title": "Pets",
-      "type": "array",
-      "items": {
-        "type": "string"
-      }
-    }
```
Comment (dmontagu) [inline docs/examples/custom_root_field.json]: This was almost completely duplicated in the python example already; it seemed simpler to just put it there in its entirety. Let me know if you prefer to keep this as a separate file though.

Comment (samuelcolvin) [reply docs/examples/custom_root_field.json]: yes, much better as one file.

File: docs/index.rst [hunk 3]
```diff
@@ -1018,7 +1017,9 @@ Example:
 ``model.json(...)``
 ~~~~~~~~~~~~~~~~~~~
 
-The ``json()`` method will serialise a model to JSON, ``json()`` in turn calls ``dict()`` and serialises its result.
+The ``json()`` method will serialise a model to JSON. Typically, ``json()`` in turn calls ``dict()`` and
+serialises its result. (For models with a :ref:`custom root <custom_root>`, after calling ``dict()``,
```
Comment (samuelcolvin) [inline docs/index.rst]:
```suggestion
serialises its result. (For models with a :ref:`custom root type <custom_root>`, after calling ``.dict()``,
```

File: docs/index.rst [hunk 4]
```diff
@@ -1018,7 +1017,9 @@ Example:
 ``model.json(...)``
 ~~~~~~~~~~~~~~~~~~~
 
-The ``json()`` method will serialise a model to JSON, ``json()`` in turn calls ``dict()`` and serialises its result.
+The ``json()`` method will serialise a model to JSON. Typically, ``json()`` in turn calls ``dict()`` and
```
Comment (samuelcolvin) [inline docs/index.rst]:
```suggestion
The ``.json()`` method will serialise a model to JSON. Typically, ``.json()`` in turn calls ``.dict()`` and
```

I think we should use `.dict()` in this whole section to distinguish from `dict(model)`, and therefore `.json()` to be consistent.

File: pydantic/main.py [hunk 5]
```diff
@@ -329,11 +329,10 @@ def json(
         `encoder` is an optional function to supply as `default` to json.dumps(), other arguments as per `json.dumps()`.
         """
         encoder = cast(Callable[[Any], Any], encoder or self._json_encoder)
-        return json.dumps(
-            self.dict(include=include, exclude=exclude, by_alias=by_alias, skip_defaults=skip_defaults),
-            default=encoder,
-            **dumps_kwargs,
-        )
+        unstructured = self.dict(include=include, exclude=exclude, by_alias=by_alias, skip_defaults=skip_defaults)
```
Comment (samuelcolvin) [inline pydantic/main.py]: slightly weird name, surely just `data`?

Comment (dmontagu) [reply pydantic/main.py]: a holdover from my attrs/cattrs days

**Pull request merged**